src/Workspaces/Core/Portable/Utilities/IDictionaryExtensions.cs

```diff
@@ -42,6 +43,16 @@ internal static class IDictionaryExtensions
             collection.Add(value);
         }
 
+        public static void MultiAdd<TKey, TValue>(this IDictionary<TKey, ImmutableArray<TValue>> dictionary, TKey key, TValue value)
+        {
+            if (!dictionary.TryGetValue(key, out var collection))
+            {
+                collection = ImmutableArray<TValue>.Empty;
+            }
+
+            dictionary[key] = collection.Add(value);
```

Comment:
Did we lose the special case for the single-element sharing though?

Reply:
Oh, yeah, right. Updated.